jdk/test/lib/testlibrary/jdk/testlibrary/Utils.java

```diff
@@ -464,4 +467,782 @@ public static Path createTempDirectory(String prefix, FileAttribute<?>... attrs)
+     * @param e5 the fifth element
+     * @return a {@code Set} containing the specified elements
+     * @throws IllegalArgumentException if there are any duplicate elements
+     * @throws NullPointerException if an element is {@code null}
+     */
+    public static <E> Set<E> setOf(E e1, E e2, E e3, E e4, E e5) {
+        if (e1 == null) { throw new NullPointerException("e1"); }
+        if (e2 == null) { throw new NullPointerException("e2"); }
+        if (e3 == null) { throw new NullPointerException("e3"); }
+        if (e4 == null) { throw new NullPointerException("e4"); }
+        if (e5 == null) { throw new NullPointerException("e5"); }
+
+        Set<E> s = new HashSet<>(10);
+        boolean added = s.add(e1);
+        if (!added) { throw new IllegalArgumentException("duplicate 1"); }
+        added = s.add(e2);
+        if (!added) { throw new IllegalArgumentException("duplicate 2"); }
+        added = s.add(e3);
+        if (!added) { throw new IllegalArgumentException("duplicate 3"); }
+        added = s.add(e4);
+        if (!added) { throw new IllegalArgumentException("duplicate 4"); }
+        added = s.add(e5);
+        if (!added) { throw new IllegalArgumentException("duplicate 5"); }
+
+        return Collections.unmodifiableSet(s);
+    }
+
+    /**
+     * Returns an unmodifiable set containing six elements.
+     *
+     * @param <E> the {@code Set}'s element type
+     * @param e1 the first element
+     * @param e2 the second element
+     * @param e3 the third element
+     * @param e4 the fourth element
+     * @param e5 the fifth element
+     * @param e6 the sixth element
+     * @return a {@code Set} containing the specified elements
+     * @throws IllegalArgumentException if there are any duplicate elements
+     * @throws NullPointerException if an element is {@code null}
+     */
+    public static <E> Set<E> setOf(E e1, E e2, E e3, E e4, E e5, E e6) {
+        if (e1 == null) { throw new NullPointerException("e1"); }
+        if (e2 == null) { throw new NullPointerException("e2"); }
+        if (e3 == null) { throw new NullPointerException("e3"); }
+        if (e4 == null) { throw new NullPointerException("e4"); }
+        if (e5 == null) { throw new NullPointerException("e5"); }
+        if (e6 == null) { throw new NullPointerException("e6"); }
+
+        Set<E> s = new HashSet<>(10);
+        boolean added = s.add(e1);
+        if (!added) { throw new IllegalArgumentException("duplicate 1"); }
+        added = s.add(e2);
+        if (!added) { throw new IllegalArgumentException("duplicate 2"); }
+        added = s.add(e3);
+        if (!added) { throw new IllegalArgumentException("duplicate 3"); }
+        added = s.add(e4);
+        if (!added) { throw new IllegalArgumentException("duplicate 4"); }
+        added = s.add(e5);
+        if (!added) { throw new IllegalArgumentException("duplicate 5"); }
+        added = s.add(e6);
+        if (!added) { throw new IllegalArgumentException("duplicate 6"); }
```

Comment:
Could be shortened, but thats up to you (if one does not care for the string, a simple `if (!(s.add(e1) && s.add(e2) && ...` would be sufficient.

Reply:
Not sure why I wrote this in such a long-winded way. Maybe there was more to it at one point and I didn't refine it after making further changes :/

Anyway, converted to a single line but kept the message, which can be useful in debugging (this is test output after all)